src/main/clojure/jarl/builtins/array.clj

```diff
@@ -15,41 +15,27 @@
 ;
 
 (ns jarl.builtins.array
-  (:import (se.fylling.jarl BuiltinException)
-           (clojure.lang PersistentVector)))
+  (:require [jarl.builtins.utils :refer [check-args]]))
 
-; TODO: Move somewhere else for reuse
-(defn java->rego [value]
-  (condp instance? value
-    String "string"
-    Boolean "boolean"
-    Double "floating-point number"
-    Float "floating-point number"
-    Integer "number"
-    Long "number"
-    Number "number"
-    PersistentVector "array"
-    (str "unknown type: " (type value) " from value: " value)))
-
-(defn builtin-concat [a b]
-  (when-not (vector? a)
-    (throw (BuiltinException. (format "array.concat: operand 1 must be array but got %s", (java->rego a)))))
-  (when-not (vector? b)
-    (throw (BuiltinException. (format "array.concat: operand 2 must be array but got %s", (java->rego b)))))
+(defn builtin-concat
+  "Implementation of array.concat built-in"
+  {:builtin "array.concat" :args-types ["array" "array"]}
+  [a b]
+  (check-args (meta #'builtin-concat) a b)
```

Comment:
Is it possible to pull out the `meta` of func from the outside, where it's being called? If so, we could maybe move this to `jarl.eval/eval-builtin-func`

Reply:
Yes, that's definitely doable! Sounds reasonable to move the args check there, but I'd need to rewrite the unit tests too for error handling. Will save for a later PR.